views/reducers/Character.js

```diff
@@ -1,19 +1,32 @@
 import ACTIONS from '../actions';
-import { nextPosition } from './Movement.js';
+import { nextPositionToPoint } from './Movement.js';
 
 const Three = require('three');
 
-export const characterInitialState = {
+const InitialState = {
   characterPosition: new Three.Vector3(0, 0, 0),
+  characterRotation: new Three.Euler(0, 0, 0, 'XYZ'),
 };
 
-export const characterReducer = (state = characterInitialState, action) => {
+const Character = (state = InitialState, action) => {
   switch (action.type) {
     case ACTIONS.CHARACTER.WALK:
+      const nextPosition = nextPositionToPoint(state.characterPosition, action);
+      const fromV = new Three.Vector3(-1, 0, 0);
+      const toV = state.characterPosition.clone().sub(nextPosition).normalize();
+      toV.set(-toV.z, 0, toV.x);
+      /*
+      * OK, so this works, but its a hack. We are doing something wrong here but what?
+      * */
+      const rotation = new Three.Euler().setFromQuaternion(
+        new Three.Quaternion().setFromUnitVectors(fromV, toV).normalize(), 'XYZ');
```

Comment:
You shouldn't be declaring stuff inside a case, also you're clearly doing too much stuff here. I'd suggest you compute this stuff into the action due you maybe blocking the state.